app/src/main/java/com/example/playlistmaker/SearchActivity.kt

```diff
@@ -76,58 +97,56 @@ class SearchActivity : AppCompatActivity() {
                     if (response.code() == 200) {
 
                         if (response.body()?.resultCount == 0) {
-                            trackNotFoundVidget.setVisibility(View.VISIBLE)
+                            binding.progressBar.visibility = View.GONE
+                            binding.trackNotFoundVidget.setVisibility(View.VISIBLE)
                         } else {
                             trackList.clear()
                             trackList.addAll(response.body()?.results!!)
                             trackAdapter.tracks = trackList
                             trackAdapter.notifyDataSetChanged()
-                            recycleViewTracks.setVisibility(View.VISIBLE)
+                            binding.progressBar.visibility = View.GONE
+                            binding.recycleViewTracks.setVisibility(View.VISIBLE)
 
                         }
 
                     } else {
-                        noInternetVidget.setVisibility(View.VISIBLE)
+                        binding.progressBar.visibility = View.GONE
+                        binding.noInternetVidget.setVisibility(View.VISIBLE)
                     }
                 }
 
                 override fun onFailure(call: Call<TrackResponse>, t: Throwable) {
                     Log.d("SEARCH_LOG", "FAIL")
-                    noInternetVidget.setVisibility(View.VISIBLE)
+                    binding.noInternetVidget.setVisibility(View.VISIBLE)
                 }
             })
-
+        }
     }
 
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_search)
 
-        trackNotFoundVidget = findViewById<View>(R.id.not_found_widget)
-        noInternetVidget = findViewById<View>(R.id.network_problem_widget)
+        binding = ActivitySearchBinding.inflate(layoutInflater)
+        val view = binding.root
+        setContentView(view)
```

Comment:
🍏 Переменная view используется один раз, можно не сохранять отдельно.
```suggestion
        setContentView(binding.root)
```

Reply:
Исправила